### PR TITLE
ZonedDateTime.toLocaleString() and last cookbook example

### PR DESCRIPTION
**Pull request by ptomato**
See: #569

### CODE REVIEW

File: polyfill/lib/intl.mjs [hunk 1]
```diff
@@ -319,15 +358,39 @@ function extractOverrides(temporalObj, main) {
...
+      instant: main[TZ_RESOLVED].getInstantFor(datetime),
       formatter: main[DATETIME]
     };
   }
 
+  if (ES.IsTemporalZonedDateTime(temporalObj)) {
+    const calendar = GetSlot(temporalObj, CALENDAR);
+    if (calendar.id !== 'iso8601' && calendar.id !== main[CAL_ID]) {
+      throw new RangeError(
+        `cannot format ZonedDateTime with calendar ${calendar.id} in locale with calendar ${main[CAL_ID]}`
```
Comment (Ms2ger) [inline polyfill/lib/intl.mjs:370]: Use `CalendarToString` (and make sure to call it just once.)

File: spec/intl.html [hunk 2]
```diff
@@ -315,11 +315,16 @@ <h1>PartitionDateTimePattern ( _dateTimeFormat_, _x_ )</h1>
             1. <ins>Let _pattern_ be _dateTimeFormat_.[[TemporalDateTimePattern]].</ins>
             1. <ins>Let _tm_ be { [[weekday]]: ! ToDayOfWeek(_x_.[[ISOYear]], _x_.[[ISOMonth]], _x_.[[ISODay]]), [[year]]: _x_.[[ISOYear]], [[month]]: _x_.[[ISOMonth]], [[day]]: _x_.[[ISODay]], [[hour]]: _x_.[[ISOHour]], [[minute]]: _x_.[[ISOMinute]], [[second]]: _x_.[[ISOSecond]] }.</ins>
           1. <ins>If _date_ has an [[InitializedTemporalZonedDateTime]] internal slot, then</ins>
-            1. <ins>Let _calendar_ be ? CalendarToString(_x_).</ins>
+            1. <ins>Let _timeZone_ be ? TimeZoneToString(_x_.[[TimeZone]]).</ins>
+            1. <ins>If _dateTimeFormat_.[[TimeZone]] is not equal to DefaultTimeZone(), and _timeZone_ is not equal to _dateTimeFormat_.[[TimeZone]], then</ins>
+              1. <ins>Throw a *RangeError* exception.</ins>
+            1. <ins>Let _calendar_ be ? CalendarToString(_x_.[[Calendar]]).</ins>
             1. <ins>If _calendar_ is not *"iso8601"* and not equal to _dateTimeFormat_.[[Calendar]], then</ins>
               1. <ins>Throw a *RangeError* exception.</ins>
```
Comment (Ms2ger) [inline spec/intl.html:323]: Is the polyfill doing these checks in the opposite order?

**Review by Ms2ger — CHANGES_REQUESTED** [review]

File: polyfill/lib/intl.mjs (hunk 1 above)
Comment (ptomato) [reply polyfill/lib/intl.mjs]: Good point, I've added another commit to fix the existing instances of this problem as well.

File: docs/cookbook/meetingPlanner.js [hunk 3]
```diff
@@ -1,46 +1,43 @@
...
+    cell.className = `time-${columnTime.hour}`;
 
     // Highlight the current hour in each row
-    if (hours === calendarNow.hour) cell.className += ' time-current';
+    if (hours === now.hour) cell.className += ' time-current';
 
     // Show the date in midnight cells
     let formatOptions;
-    if (dt.hour === 0) {
+    if (columnTime.hour === 0) {
```
Comment (justingrant) [inline docs/cookbook/meetingPlanner.js:35]: I think you probably want to compare columnTime to `.startOfDay` here, not zero, if you want to handle cases where DST skips an hour starting at midnight.

**Review by justingrant — APPROVED** [review]
Looks good. I admittedly don't know much about the innards of toLocaleString so I focused my review on the cookbook sample and the high-level expectations like throwing if zones are mismatched.

**Review by Ms2ger — APPROVED** [review]

File: docs/cookbook/meetingPlanner.js (hunk 3 above)
Comment (ptomato) [reply docs/cookbook/meetingPlanner.js:35]: That's a good point, it does handle other skipped and repeated times gracefully, so it should here as well.

**Pull request merged**